community/kernel/src/main/java/org/neo4j/helpers/Listeners.java

```diff
@@ -31,68 +31,56 @@
 {
     public interface Notification<T>
     {
-        void notify(T listener);
+        void notify( T listener );
     }
 
-    public static <T> Iterable<T> newListeners()
+    public static <T> Collection<T> newListeners()
     {
-        return new LinkedList<T>();
+        return new LinkedList<>();
```

Comment:
WDYT about changing `LinkedList` to `ArrayList` everywhere? `LinkedList` seem to be useless because listeners are copied on every modification and main operation is traversal.